presto-manager-controller/src/main/java/com/teradata/prestomanager/controller/api/ControllerConfigAPI.java

```diff
@@ -0,0 +1,218 @@
+                .entity(Entity.entity(url, MediaType.TEXT_PLAIN))
+                .build();
+
+        return getResponseFromRequest(apiScope, apiRequester, nodeId);
+    }
+
+    @DELETE
+    @Path("/{file}")
+    public Response deleteConfigFile(
+            @PathParam("file") @ApiParam("The name of a file") String file,
+            @QueryParam("scope") @DefaultValue("INVALID") String scope,
+            @QueryParam("nodeId") @DefaultValue("null") String nodeId)
+    {
+        ApiScope apiScope = ApiScope.fromString(scope);
+        try{
+            checkArguments(apiScope, nodeId);
+        }
+        catch (IllegalArgumentException e) {
+            return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("Invalid scope").build();
+        }
+
+        ApiRequester apiRequester = ApiRequester.builder(ControllerConfigAPI.class)
+            .pathMethod("deleteConfigFile")
+            .httpMethod(HttpMethod.DELETE)
+            .resolveTemplate("file", file)
+            .accept(MediaType.TEXT_PLAIN)
+            .build();
+
+        return getResponseFromRequest(apiScope, apiRequester, nodeId);
+    }
+
+    private void checkArguments (ApiScope scope, String nodeId)
+    {
+        if ((nodeId != null && scope != ApiScope.INVALID) || (nodeId == null && scope == ApiScope.INVALID)) {
+            LOGGER.error("Invalid scope");
+            throw new IllegalArgumentException("Invalid scope");
+        }
+    }
+
+    private Response getResponseFromRequest(ApiScope apiScope, ApiRequester apiRequester, String nodeId)
+    {
+        switch (apiScope) {
+            case CLUSTER:
+            case WORKERS:
+                List<Response> responseList = new ArrayList<>();
+                for (URI uri : nodeSet.getUris(apiScope.name())) {
+                    responseList.add(apiRequester.send(uri));
+                }
+                return Response.status(ExtendedStatus.MULTI_STATUS)
+                        .entity(new Gson().toJson(responseList)).type(MediaType.APPLICATION_JSON).build();
+            case COORDINATOR:
+                Collection<URI> coordinators = nodeSet.getUris(apiScope.name());
+                if (coordinators.size() != 1) {
+                    LOGGER.error("Wrong number of coordinator");
+                    return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("Wrong number of coordinator").build();
+                }
+                return apiRequester.send(coordinators.iterator().next());
+            case INVALID:
+                URI uri = nodeSet.getUriById(Integer.parseInt(nodeId));
+                if (uri == null) {
+                    LOGGER.error("Invalid node ID");
+                    return Response.status(Response.Status.INTERNAL_SERVER_ERROR).entity("Invalid node ID").build();
+                }
+                return apiRequester.send(uri);
+            default:
+                LOGGER.fatal("Impossible scope, Reach unreachable point");
```

Comment:
The error message should be improved, yes, but HTTP 500 is the correct response; at the time this is called, the scope should already have been checked; if we get here, the program's logic is wrong.